website/addons/dataverse/model.py

```diff
@@ -88,22 +94,45 @@ class AddonDataverseNodeSettings(AddonNodeSettingsBase):
 
     dataverse_alias = fields.StringField()
     dataverse = fields.StringField()
-    study_hdl = fields.StringField()
-    study = fields.StringField()
+    dataset_doi = fields.StringField()
+    _dataset_id = fields.StringField()
+    dataset = fields.StringField()
+
+    # Legacy fields
+    study_hdl = fields.StringField()    # Now dataset_doi
+    study = fields.StringField()        # Now dataset
 
     user_settings = fields.ForeignField(
         'addondataverseusersettings', backref='authorized'
     )
 
     @property
-    def is_fully_configured(self):
-        return bool(self.has_auth and self.study_hdl is not None)
+    def dataset_id(self):
```

Comment:
Not sure how I feel about a property that does such expensive work (HTTP and writing to database). I see the motivation but wonder if there are better alternatives.